app/routes/discover.js

```diff
@@ -20,7 +20,16 @@ export default Ember.Route.extend(Analytics, ResetScrollMixin, {
     model() {
         return this
             .get('store')
-            .findAll('preprint-provider', { reload: true });
+            .query('preprint-provider', { reload: true }).then((results) => {
+                return {
+                    preprintProviders: results,
+                    meta: results.get('meta')
+                };
+            });
+    },
+    setupController(controller, { preprintProviders, meta }) {
+        this._super(controller, preprintProviders);
+        controller.set('meta', meta);
```

Comment:
Is this necessary? Wouldn't you have access to this in the controller through `model.meta`?

Reply:
According to Ember official guide, to access top-level metadata outside of model hook we have to set it for the controller. (https://guides.emberjs.com/v2.17.0/models/handling-metadata/)

Reply:
Had never seen this, and assumed that it worked like RSVP.hash, where the model became an object and meta would just be another key in it.